clang/lib/StaticAnalyzer/Checkers/FixedAddressChecker.cpp

```diff
@@ -63,11 +62,39 @@ void FixedAddressChecker::checkPreStmt(const BinaryOperator *B,
         "Using a fixed address is not portable because that address will "
         "probably not be valid in all environments or platforms.";
     auto R = std::make_unique<PathSensitiveBugReport>(BT, Msg, N);
-    R->addRange(B->getRHS()->getSourceRange());
+    R->addRange(SrcExpr->getSourceRange());
     C.emitReport(std::move(R));
   }
 }
 
+void FixedAddressChecker::checkPreStmt(const BinaryOperator *B,
+                                       CheckerContext &C) const {
+  if (B->getOpcode() != BO_Assign)
+    return;
+
+  checkUseOfFixedAddress(B->getType(), B->getRHS(), C);
+}
+
+void FixedAddressChecker::checkPreStmt(const DeclStmt *D,
+                                       CheckerContext &C) const {
+  for (const auto *D1 : D->decls()) {
+    if (const auto *VD1 = dyn_cast<VarDecl>(D1); VD1 && VD1->hasInit())
+      checkUseOfFixedAddress(VD1->getType(), VD1->getInit(), C);
+  }
+}
+
+void FixedAddressChecker::checkPreStmt(const CallExpr *CE,
+                                       CheckerContext &C) const {
+  const FunctionDecl *Callee = CE->getDirectCallee();
+  if (!Callee)
```

Comment:
What if the callee is a function pointer, could you add a test? I suppose, this check should also check such callexprs, or at least I cannot see a reason why it couldn't.